code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm

```diff
@@ -199,6 +199,8 @@
 		target_turfs += current_turfs
 		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(current_turfs, 2)
 
+	var/can_heal = !(SEND_SIGNAL(xeno, COMSIG_XENO_PRE_HEAL) & COMPONENT_CANCEL_XENO_HEAL)
```

Comment:
This is a bad signal.

Signals should only be sent from one location in code. If they need to be sent from multiple locations in code, they need to be separated into their own signal.

Consider isolating this into its own `can_heal` proc on the xenomorph type itself and calling the signal from there.

Reply:
I notice this signal is sent from multiple places in code right now, but that's bad. I think you should just do this basic refactor since it'll prevent headache in the future.